api/v1alpha1/externaldns_types.go

```diff
@@ -103,13 +104,17 @@ type ExternalDNSIdentity struct {
 }
 
 type ExternalDNSFilters struct {
-	// GatewayLabels contains key-value pairs that the ExternalDNS controller will use to filter the Gateways that it manages.
+	// GatewayLabelSelector is the label selector that the ExternalDNS controller will use to filter the Gateways that it manages.
 	// +optional
-	GatewayLabels map[string]string `json:"gatewayLabels,omitempty"`
+	// +nullable
+	// +kubebuilder:validation:Pattern=`^[^=]+=[^=]+$`
+	GatewayLabelSelector string `json:"gatewayLabels,omitempty"`
 
-	// RouteAndIngressLabels contains key-value pairs that the ExternalDNS controller will use to filter the HTTPRoutes and Ingresses that it manages.
+	// RouteAndIngressLabelSelector is the label selector that the ExternalDNS controller will use to filter the HTTPRoutes and Ingresses that it manages.
 	// +optional
-	RouteAndIngressLabels map[string]string `json:"routeAndIngressLabels,omitempty"`
+	// +nullable
+	// +kubebuilder:validation:Pattern=`^[^=]+=[^=]+$`
+	RouteAndIngressLabelSelector string `json:"routeAndIngressLabels,omitempty"`
```

Comment:
why not just make this a pointer? That's convention over this nullable thing

Reply:
done